js/accordionModel.js

```diff
@@ -7,60 +7,41 @@ define([
         defaults: function() {
             return _.extend({
                 _shouldCollapseItems: true,
-                _toggleSpeed: 200,
-                _activeItems: []
+                _toggleSpeed: 200
             }, ItemsModel.prototype.defaults);
         },
 
         initialize: function() {
-            if (this.get('_shouldCollapseItems') === false) {
-                this.set('_activeItems', []);
-            }
-
-            _.each(this.get('_items'), function(item) {
-                item._isActive = false;
-            });
+            this.resetActiveItems(false);
 
             ItemsModel.prototype.initialize.apply(this, arguments);
         },
 
         toggleActiveItems: function(itemIndex) {
-            var activeItems = this.get('_activeItems');
-            var index = _.indexOf(activeItems, itemIndex);
+            var activeItemsIndexes = this.getActiveItemsIndexes();
 
-            if (this.get('_shouldCollapseItems')) {
-                activeItems = (index >= 0) ? [] : [itemIndex];
+            if (activeItemsIndexes.length === 0) {
+                this.setItemAtIndexAsActive(itemIndex, false);
             } else {
-                // multiple active items possible
-                if (index >= 0) {
-                    // item is active remove it from the acite items list 
-                    activeItems.splice(index, 1);
+                if (this.get('_shouldCollapseItems')) {
+                    if (activeItemsIndexes[0] === itemIndex) {
+                        this.setItemAtIndexAsInactive(itemIndex, false);
+                    } else {
+                        this.setItemAtIndexAsActive(itemIndex, false);
+                        this.setItemAtIndexAsInactive(activeItemsIndexes[0], false);
```

Comment:
if you switch line 30 and 31 around you can drop line 42 and drop all of the trigger = false except on here:
```
this.setItemAtIndexAsInactive(activeItemsIndexes[0], false);
```